debian/control

```diff
@@ -12,7 +12,7 @@ Standards-Version: 4.1.1
 
 Package: nest
 Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}, python3, libltdl-dev,
+Depends: ${shlibs:Depends}, ${misc:Depends}, python3, libboost-all-dev, libltdl-dev,
```

Comment:
I agree with @terhorstd  and would also rather go for a minimal set of dependencies.

Reply:
@jougs  You mean build NEST with `-Dwith-mpi=OFF `and `-Dwith-boost=OFF`? This would minimize the dependencies.
@terhorstd, what do you think about this?

Reply:
building 2.20 with-boost=OFF gives a significant performance hit, so I would suggest against building without it. It really strongly depends on what the deb is going to be used for. If a user builds it to install on his own machine, I'd say with-mpi=OFF is fine. If the deb is built by an admin to roll it out on the cluster we should maybe have a more differentiated explanation. In either case, we should use the minimal with-boost=ON dependencies like libboost-test-dev, libboost-container-dev, libboost-tools-dev, etc. (not sure if these ones suffice, I may be missing a packet)

Reply:
> we should use the minimal with-boost=ON dependencies like libboost-test-dev, libboost-container-dev, libboost-tools-dev, etc. (not sure if these ones suffice, I may be missing a packet)

@jougs Are these dependencies for boost okay? Or is something missing?

Reply:
I agree with what @terhorstd said in his last comment.

@hakonsbm: can you comment on the list of boost-related libraries we need? I was not involved in that and thus can't say a whole lot about it.

Reply:
@jougs We need Boost for sorting and to compile the C++ tests. There is a separate `libboost-test` package, but I don't find a separate package for the sorting. So for a package based installation I think `libboost-all-dev` is the only option.

However, when #1502 is merged it will also be possible to download the Boost headers only and compile with those.

Reply:
@hakonsbm Thanks!
Now we have `-Dwith-mpi=OFF` and `libboost-all-dev` stays as it is.

Reply:
@terhorstd what's your opinion?

Reply:
On Travis we use separate boostlibs and those are the tested and obviously working ones. I think we should take those and not force libboost-all-dev. The sorting is probably in the minimal `libboost-dev` and not in a separate `libboost-*-dev`

Reply:
After discussion, we decided to split this to a separate issue